ca/certificate-authority-data.go

```diff
@@ -51,30 +50,11 @@ func (cadb *CertificateAuthorityDatabaseImpl) Begin() (*gorp.Transaction, error)
 // it in the database before returning. There must be an active transaction to
 // call this method. Callers should Begin the transaction, call this method,
 // perform any other work, and Commit at the end once the certificate is issued.
-func (cadb *CertificateAuthorityDatabaseImpl) IncrementAndGetSerial(tx *gorp.Transaction) (val int64, err error) {
-	if tx == nil {
-		err = fmt.Errorf("No transaction given")
-		return
-	}
-
-	rowObj, err := tx.Get(SerialNumber{}, 1)
-	if err != nil {
-		return
-	}
-
-	row, ok := rowObj.(*SerialNumber)
-	if !ok {
-		err = fmt.Errorf("No serial number found. This is a serious issue")
-		return
-	}
-
-	val = row.Number
-	row.Number = val + 1
-
-	_, err = tx.Update(row)
+func (cadb *CertificateAuthorityDatabaseImpl) IncrementAndGetSerial(tx *gorp.Transaction) (int64, error) {
+	r, err := tx.Exec("REPLACE INTO serialNumber (stub) VALUES ('a');")
 	if err != nil {
-		return
+		return -1, err
```

Comment:
It was a return value that couldn't even accidentally be used as an index or similar. 0 would have worked just fine and it was a flip of a coin in my head. We can adjust it if you'd rather use a zero value.